src/styles/theme.ts

```diff
@@ -13,6 +13,10 @@ const theme = createTheme({
     error: {
       main: red.A400,
     },
+    ActionButton: {
+      bgColor: '#007bff',
+      color: '#ffffff',
+    },
```

Comment:
ActionButton is not a known property in 'PaletteOptions' due to which the checks have failed. Another way needs to be figured out to set the button text color

Reply:
Yeah good call @DivyanshuParwal. As an example I actually did [typescript module augmentation in orcasite](https://github.com/orcasound/orcasite/blob/master/ui/src/styles/theme.ts#L7-L22).

But even better than making `ActionButton` a palette color would be to add a more generic name (assuming the color will be re-used), e.g. `accent1` like I did for orcasite.

The reason is that if we just add each components colors to the theme like this (name every color after the component that uses it), then it kind of defeats the purpose of the theme (which is, color consistency and reusability). At that point it's better to just set the colors one-off directly inside the component using the `sx prop` or something like that, because it's less work for the same outcome.

So if we think we won't re-use this color (`#007bff`) then it's best to add the style locally in the component itself. If it'll be re-used, then the way it's done in this PR is good, just would re-name the color and add the typescript stuff so ti compiles.

Finally, for the text color, I would just use `contrastText` which should be [calculated automatically](https://mui.com/customization/palette/#providing-the-colors-directly) from the `main` color

Reply:
Would the implementation be recommended on its own createTheme() or in the existing theme const?

Reply:
I would try to keep everything in one theme if possible. The more themes we have the less useful they are (the whole point is to have consistency across the entire app)

Reply:
Looks like I accidentally implemented the exact same concept I went with last time. Will refactor and try again.